### PR TITLE
Chore: Fix golangci-lint timeout config, disable govet

### PR DESCRIPTION
**Pull request by aknuds1**
**What this PR does / why we need it**:
Fix the configuration of golangci-lint by configuring _timeout_ instead of _deadline_, since the former is the [correct](https://github.com/golangci/golangci-lint#config-file) name of this parameter.

Also disable govet among its linters, since we run it separately.

:crossed_fingers: this fixes our CI issues.

### CODE REVIEW

**Review by papagian — APPROVED** [review]
LGTM

**Review by marefr — APPROVED** [review]
LGTM

But can it actually run longer than 10 minutes? Feels like a long time...

**Comment by aknuds1** [comment]
@marefr I don't know if there's any real need to increase the timeout, just to be on the safe side.

**Comment by aknuds1** [comment]
It got killed again :/ Is there a way to tell for sure if it OOMs or not?

**Comment by bergquist** [comment]
Seems like the timeout comes from the listing tool. 

> Too long with no output (exceeded 10m0s): context deadline exceeded

That's weird cause 10m should be more than enough to lint Grafanas codebase.

**Comment by aknuds1** [comment]
> Seems like the timeout comes from the listing tool.
> 
> > Too long with no output (exceeded 10m0s): context deadline exceeded
> 
> That's weird cause 10m should be more than enough to lint Grafanas codebase.

@bergquist That's only the latest timeout, which is enforced by CircleCI. I just increased it to 20 minutes. The reason that linting takes longer now is probably because I adjust concurrency to 1 and made garbage collection more aggressive (GOGC=20).

**Comment by aknuds1** [comment]
It keeps getting killed. Maybe a solution is to run golangci-lint in two steps, 50% of the linters each time.

File: .circleci/config.yml [hunk 1]
```diff
@@ -721,16 +721,22 @@ jobs:
       - checkout
       - run:
           name: Install Go linters
-          command: make scripts/go/bin/golangci-lint scripts/go/bin/revive scripts/go/bin/gosec
+          command: |
+            pushd /tmp
+            curl -fLO https://github.com/golangci/golangci-lint/releases/download/v1.24.0/golangci-lint-1.24.0-linux-amd64.tar.gz
```
Comment (bergquist) [inline .circleci/config.yml]: Sounds reasonable. But once this is working we need to figure out how CI runs the same version as devs to locally.

Comment (aknuds1) [reply .circleci/config.yml]: @bergquist Sure. The linting time has been reduced to ~3 minutes in CI, so I suspect using the official build is what helped.

**Pull request merged**